nixos/modules/system/boot/plymouth.nix

```
@@ -167,12 +167,22 @@ in
     systemd.services.plymouth-poweroff.wantedBy = [ "poweroff.target" ];
     systemd.services.plymouth-reboot.wantedBy = [ "reboot.target" ];
     systemd.services.plymouth-read-write.wantedBy = [ "sysinit.target" ];
+    systemd.services.plymouth-switch-root-initramfs.wantedBy = [
+      "halt.target"
+      "kexec.target"
+      "poweroff.target"
+      "reboot.target"
+      "generate-shutdown-ramfs.service"
+      "plymouth-switch-root-initramfs.service"
+    ];
     systemd.services.systemd-ask-password-plymouth.wantedBy = [ "multi-user.target" ];
     systemd.paths.systemd-ask-password-plymouth.wantedBy = [ "multi-user.target" ];
 
     # Prevent Plymouth taking over the screen during system updates.
     systemd.services.plymouth-start.restartIfChanged = false;
 
+    systemd.shutdownRamfs.storePaths = [ "${plymouth}/libexec/plymouth/plymouthd-fd-escrow" ];
```

```suggestion
    # helper binary which is used to hold onto the pixel-displays fds until the end

    systemd.shutdownRamfs.storePaths = [ "${plymouth}/libexec/plymouth/plymouthd-fd-escrow" ];
```
I am very insecure about the other changes but I cherry-pick them into my system